### DISCUSSION TITLE
Add meta-data to DLQ

### DISCUSSION THREAD

**Discussion by rocketraman** [apache/pulsar#18840] [General]
**Is your feature request related to a problem? Please describe.**
The automatic DLQ is a nice feature, but its lacking any ability to add meta-data to the entry in the DLQ. For example, setting a property like "causalException" with an exception message and stack trace is very useful when looking at the entries in the DLQ and deciding how to handle them.

**Describe the solution you'd like**
One possible solution may be to allow passing additional properties to the `negativeAcknowledge` method. When an entry is created in the DLQ, the additional props provided would be added to the DLQ message properties.

**Alternatives**
Manual DLQ handling. Use Pulsar's redelivery count to manually produce to a DLQ topic, along with the necessary meta-data. ~This implies that the redelivery count is populated in the `Message` regardless of the presence of a `DeadLetterPolicy`. I haven't checked whether that is the case or not.~

**Comment by sijie** [comment]
@codelipenghui @congbobo184 this is a good feature to add to DLQ

**Comment by codelipenghui** [comment]
Sorry for the late response, i will take a look soon.

**Comment by jefferyshivers-toast** [comment]
Has there been any movement on this? This is an example of a similar feature in Azure Service Bus: https://docs.microsoft.com/en-us/azure/service-bus-messaging/service-bus-dead-letter-queues#moving-messages-to-the-dlq.
